tty.c

```diff
@@ -2385,6 +2404,54 @@ tty_check_bg(struct tty *tty, struct window_pane *wp, struct grid_cell *gc)
 		gc->bg -= 90;
 }
 
+static void
+tty_check_us(struct tty *tty, struct window_pane *wp, struct grid_cell *gc)
+{
+	u_char	r, g, b;
+	u_int	colours;
+	int	c;
+
+	/* Perform substitution if this pane has a palette. */
+	if (~gc->flags & GRID_FLAG_NOPALETTE) {
+		if ((c = window_pane_get_palette(wp, gc->us)) != -1)
+			gc->us = c;
+	}
+
+	/* Check if this is a 24-bit colour. */
+	if (gc->us & COLOUR_FLAG_RGB) {
+		/* Not a 24-bit terminal? Translate to 256-colour palette. */
+		if (!tty_term_has(tty->term, TTYC_SETRGBU)) { // TODO: kmo
+			colour_split_rgb(gc->us, &r, &g, &b);
+			gc->us = colour_find_rgb(r, g, b);
+		} else
+			return;
+	}
+
+	/* How many colours does this terminal have? */
+	if ((tty->term->flags|tty->term_flags) & TERM_256COLOURS)
+		colours = 256;
+	else
+		colours = tty_term_number(tty->term, TTYC_COLORS);
+
+	/* Is this a 256-colour colour? */
+	if (gc->us & COLOUR_FLAG_256) {
+		/* And not a 256 colour mode? */
+		if (colours != 256) {
+			gc->us = colour_256to16(gc->us);
+			if (gc->us & 8) {
+				gc->us &= 7;
+				if (colours >= 16)
+					gc->us += 90;
+			}
+		}
+		return;
+	}
+
+	/* Is this an aixterm colour? */
```

Comment:
Since we can not specify underline color in aixterm, should this block simply be removed?

Reply:
Yes.